forum/qa-plugin/report-reason/q2apro-flag-reasons-page.php

```diff
@@ -0,0 +1,177 @@
+                    qa_db_post_parent_q_selectspec($questionId),
+                    false
+                );
+
+                list($question, $childPosts, $aChildPosts,
+                    $closePost, $duplicatePosts
+                    ) = $questionData;
+
+                $error = qa_flag_error_html($question, $userId, $questionId);
+                
+                if (!$error) {
+                    $handle   = qa_userid_to_handle($userId);
+                    $cookieId = qa_cookie_get();
+
+                    $answers         = qa_page_q_load_as($question, $childPosts);
+                    $commentsFollows = qa_page_q_load_c_follows($question, $childPosts, $aChildPosts, $duplicatePosts);
+
+                    if (qa_flag_set_tohide($question, $userId, $handle, $cookieId, $question)) {
+                        qa_question_set_status(
+                            $question, QA_POST_STATUS_HIDDEN, null, null, null, $answers, $commentsFollows, $closePost
+                        ); // hiding not really by this user so pass nulls
+                    }
+                    if($reasonId >= 0 && $reasonId <= 6) {
+                        qa_db_query_sub(
+                        '
+                            INSERT INTO `^flagreasons` (`userid`, `postid`, `reasonid`, `notice`) 
+                            VALUES (#, #, #, $)
+                        ', $userId, $postId, $reasonId, $notice
+                        );
+                    }
+                }
+            } elseif ('a' === $postType) {
+                $answerId = $postId;
+
+                list($answer, $question, $qChildPosts, $aChildPosts) = qa_db_select_with_pending(
+                    qa_db_full_post_selectspec($userId, $answerId),
+                    qa_db_full_post_selectspec($userId, $questionId),
+                    qa_db_full_child_posts_selectspec($userId, $questionId),
+                    qa_db_full_child_posts_selectspec($userId, $answerId)
+                );
+
+                $commentsFollows = qa_page_q_load_c_follows($question, $qChildPosts, $aChildPosts);
+
+                $error = qa_flag_error_html($answer, $userId, $questionId);
+
+                if (!$error) {
+                    $handle   = qa_userid_to_handle($userId);
+                    $cookieId = qa_cookie_get();
+
+                    if (qa_flag_set_tohide($answer, $userId, $handle, $cookieId, $question)) {
+                        qa_answer_set_status(
+                            $answer, QA_POST_STATUS_HIDDEN, null, null, null, $question, null//$commentsFollows
+                        ); // hiding not really by this user so pass nulls
+                    }
+
+                    qa_db_query_sub(
+                        '
+                        INSERT INTO `^flagreasons` (`userid`, `postid`, `reasonid`, `notice`) 
+                        VALUES (#, #, #, $)
+                    ', $userId, $answer['postid'], $reasonId, $notice
```

Comment:
skad pewnosc ze w answer bedzie postid? A jak nie zwróci nic?

Reply:
No zakładam że jeśli jest triggerowany event o jakimś ID to prawdopodobnie jest on w bazie, ale fakt, sprawdzę to.